src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -167,6 +167,20 @@ private void maybeInitJmx()
         }
     }
 
+    @VisibleForTesting
+    public static Runnable SPECULATION_THRESHOLD_UPDATER = 
+        () -> 
+        {
+            try
+            {
+                Keyspace.allExisting().forEach(k -> k.getColumnFamilyStores().forEach(ColumnFamilyStore::updateSpeculationThreshold));
+            }
+            catch (Throwable t)
+            {
+                logger.warn("Failed to update speculative retry thresholds.", t);
```

Comment:
after logging should call `org.apache.cassandra.utils.JVMStabilityInspector#inspectThrowable(java.lang.Throwable)`; if this is caused by OOM it will properly handle that (I don't think we touch disk so disk failure policy wouldn't happen).